### ISSUE TITLE
Buildcheck - .editorconfig seems to be ignored

### ISSUE THREAD

**Issue by JanKrivanek** [dotnet/msbuild#10684]
### Context

I came accross a case where `.editorconfig` settings for chaning checks diagnostics levels are not respected and `.editorconfig` file is not captured in the binlog - meaning it's probably not recognized at all.

### Details

Notice the `BC0201` being set to error, but output as warning. Similarly BC0203 is set to error, but not lifted from default message.

```
PS C:\tmp> dotnet --version
9.0.100-rc.2.24468.2
PS C:\tmp> type .\.editorconfig
root = true

# Buildcheck rules
[*.csproj]
build_check.BC0103.severity=error
build_check.BC0203.severity=error
build_check.BC0203.scope=work_tree_imports
build_check.BC0201.severity=error
PS C:\tmp> dotnet build /check /bl
Obnovení dokončeno (1,4s)
Používáte verzi Preview rozhraní .NET. Viz: https://aka.ms/dotnet-support-policy
  DotUtils.Calculator akce proběhla úspěšně s 1 upozorněním(i). (1,4s) → bin\Debug\net8.0\DotUtils.Calculator.dll
    C:\tmp\DotUtils.Calculator.csproj(8,4): warning BC0201: https://aka.ms/buildcheck/codes#BC0201 - Property: 'xyz' was accessed, but it was never initialized.

Sestavení akce proběhla úspěšně s 1 upozorněním(i). za 5,1s
PS C:\tmp>
```

I suspect non-US locale might have an impact here?
Relevant info from `systeminfo`:

```
OS Name:                   Microsoft Windows 10 Home
OS Version:                10.0.19045 N/A Build 19045
System Locale:             cs;Čeština
Input Locale:              cs;Čeština
```

Binlog and repro:  
[ReproAndBinlog.zip](https://github.com/user-attachments/files/17090940/ReproAndBinlog.zip)

**Comment by f-alizada** [comment]
It is related to the selected end of line selected for the file. 
https://github.com/dotnet/msbuild/blob/642eed516bbdc3888430a102aa0f6321b10becab/src/Build/BuildCheck/Infrastructure/EditorConfig/EditorConfigFile.cs#L84
We parse the string based on the current newLine of the OS. 
Since the EOF was not the one that is used on current OS if failed to parse the file.

**Comment by f-alizada** [comment]
I see two possible ways: 
1 - Identitfy the new line sequene of the file or 
2 - Try to use the  from the StreamReader (if it supports this case)

**Comment by f-alizada** [comment]
Roslyn covers the scenarios of different EOL by the `SourceText` , which was dropped during the implementation in MSBuild not to take the dependency. 
The ReadLine handles the scenario of different EOL test on LF and CLRF which could be used.

**Comment by JanProvaznik** [comment]
fixed by #10740 and merged to main from vs17.12 branch in #10749
